### PR TITLE
some long waiting cleanups and refactorings

### PR DESCRIPTION
**Pull request by dahliamalkhi**

### CODE REVIEW

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **d9aa659**.

 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #507 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/YexwHtKzwmUCGGK0k45zGGmaG8uUo34j)

File: runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java [hunk 1]
```diff
@@ -209,6 +210,13 @@ public void addTransaction(AbstractTransactionalContext tc) {
     @Override
     @SuppressWarnings("unchecked")
     public long commitTransaction() throws TransactionAbortedException {
+
+        return commitTXHelper(() -> getReadSet());
+    }
+
+    public long commitTXHelper(Supplier<Map<UUID, Set<Integer>>>
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java]: This needs to be named something better than commitTXHelper. Maybe doOptimisticCommit?

File: runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java [hunk 2]
```diff
@@ -93,38 +93,10 @@ public IStreamView copy(UUID source, UUID destination, long timestamp) {
...
-     * <p>
-     * Note: While the completion of this operation guarantees that the append
-     * has been persisted, it DOES NOT guarantee that the object has been
-     * written to the stream. For example, another client may have deleted
-     * the stream.
-     *
-     * @param object The object to append to the stream.
-     * @return The address this
-     */
-    public long acquireAndWrite(Set<UUID> streamIDs, Object object,
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java]: Is there a particular reason you removed the callbacks? It still seems to be potentially useful to know when a token was acquired and when you were overwritten?

Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java]: never used..

File: runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java [hunk 3]
```diff
@@ -130,8 +130,8 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
     private <R> R accessInner(ICorfuSMRAccess<R, T> accessMethod,
                               Object[] conflictObject, boolean isMetricsEnabled) {
         if (TransactionalContext.isInTransaction()) {
-            log.trace("Access[{}] conflictObj={} @TX[{}]", this, conflictObject,
-                    Utils.toReadableID(TransactionalContext.getCurrentContext().getTransactionID()) );
+            log.debug("Object Access[{}] conflictObj={} @TX[{}]", this, conflictObject,
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java]: is Object really necessary here? It just makes the log message unnecessarily long...

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **cdd2b75**.

 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #507 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/irxwOqpEF1IGgpV4p3AquNEw9pZUsaup)

**Pull request merged**